src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -271,57 +271,23 @@ void Engine::ApplyPerformanceHints(std::map<std::string, std::string> &config, c
-        /* performance hints set for network has higher pririty than engine ones.
-         * This applies for all the configuration parameters */
-        const auto perf_hint_name = (perf_hint != config.end())
-                                        ? PerfHintsConfig::CheckPerformanceHintValue(perf_hint->second)
-                                        : engConfig.perfHintsConfig.ovPerfHint;
-        return perf_hint_name;
-    };
-
-    const auto perf_hint_name = getPerfHintName();
-    int streams = engConfig.streamExecutorConfig._streams;
+void Engine::GetPerformanceStreams(Config& config, const std::shared_ptr<ngraph::Function>& ngraphFunc) {
+    const auto perf_hint_name = config.perfHintsConfig.ovPerfHint;
+    int streams = config.streamExecutorConfig._streams;
     if (perf_hint_name == CONFIG_VALUE(LATENCY)) {
-        streams = static_cast<int>(getAvailableNUMANodes().size());
+        streams = get_num_numa_nodes();
     } else if (perf_hint_name == CONFIG_VALUE(THROUGHPUT)) {
         streams = 0;
-    } else if (perf_hint_name.empty()) {
-        streams = streams > 1 ? streams : (engConfig.streamExecutorConfig._set_streams ? streams : 0);
     }
+    streams = config.streamExecutorConfig._streams > 0 ? config.streamExecutorConfig._streams : streams;
 
-    auto num_requests = config.find(CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS));
-    int infer_requests = 0;
-    if (num_requests != config.end()) {  // arrived with config to the LoadNetwork (and thus higher pri)
-        infer_requests = PerfHintsConfig::CheckPerformanceHintRequestValue(num_requests->second);
-    } else if (engConfig.perfHintsConfig.ovPerfHintNumRequests) {  // set thru SetConfig to the plugin, 2nd priority
-        infer_requests = engConfig.perfHintsConfig.ovPerfHintNumRequests;
-    }
+    get_num_streams(streams, ngraphFunc, config);
 
-    const auto common_hints = get_num_streams(streams, infer_requests, ngraphFunc, engConfig.streamExecutorConfig);
-
-    config[CONFIG_KEY(CPU_THROUGHPUT_STREAMS)] = common_hints.first;
-    config[ov::num_streams.name()] = common_hints.first;
-    config[CONFIG_KEY(CPU_THREADS_NUM)] = std::to_string(common_hints.second.num_threads);
-    config[CONFIG_KEY_INTERNAL(BIG_CORE_STREAMS)] = std::to_string(common_hints.second.big_core_streams);
-    config[CONFIG_KEY_INTERNAL(BIG_CORE_LOGIC_STREAMS)] = std::to_string(common_hints.second.big_core_logic_streams);
-    config[CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)] = std::to_string(common_hints.second.small_core_streams);
-    config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] = std::to_string(common_hints.second.threads_per_stream_big);
-    config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)] =
-        std::to_string(common_hints.second.threads_per_stream_small);
-    config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(common_hints.second.small_core_offset);
+    if (config.exclusiveAsyncRequests) {  // Exclusive request feature disables the streams
```

Comment:
This condition is laready handled inside readProperties: it sets `config.streamExecutorConfig._streams` equal 1 in case `config.exclusiveAsyncRequests=true`. Why do we need to duplicate it here?

Reply:
Ok, you are right. I removed it.